src/RelayPricer.ts

```diff
@@ -0,0 +1,34 @@
+import CoinBase from './api/CoinBase';
+import { SourceApi } from './types/RelayPricer';
+
+export default class RelayPricer {
+    private readonly sourceApi: SourceApi;
+
+    constructor(sourceApi: SourceApi | string) {
+        if (typeof sourceApi === 'string' || sourceApi instanceof String) {
+            switch (sourceApi as string) {
+                case 'coinbase':
+                    this.sourceApi = new CoinBase();
+                    break;
+                default:
+                    throw new Error('Source api not provided');
+            }
+        } else {
+            this.sourceApi = sourceApi as SourceApi;
+        }
+    }
+
+    async getPrice(
+        pair1: string,
+        pair2: string,
+        intermediaryPair?: string
+    ): Promise<number> {
+        const intermediary = intermediaryPair ? intermediaryPair : 'USD';
+        const price1 = await this.sourceApi.query(pair1, intermediary);
+        const price2 = await this.sourceApi.query(pair2, intermediary);
+        if (price1 === 0 || price2 === 0) {
```

Comment:
can the price ever be zero? what happens if I send in a currency that doesn't exist? also maybe instead of pair1 and pair2, we might call them baseCurrency and counterCurrency, or sourceCurrency and targetCurrency.

Reply:
Since we can use any given API, there is a chance that someone return zero if the token does not exists. At this point if the token does not exist, in the CoinBase implementation I return an error.

Reply:
done